haskell-tests/Types/PayloadSerializationSpec.hs

```diff
@@ -60,9 +60,15 @@ checkPayload spv e =
             Left err -> counterexample err False
             Right e' -> label (groupIntoSize (fromIntegral (BS.length bs))) $ e === e'
 
+-- Modify the bitmap portion of a payload. Assumes that the input bytestring
+-- is at least 24 bits long.
 modifyPayloadBitmap :: (Word16 -> Word16) -> BS.ByteString -> BS.ByteString
 modifyPayloadBitmap f bs =
-    let Right ((header, bitmap), rest) = S.runGetState ((,) <$> S.getWord8 <*> S.getWord16be) bs 0
+    let res = S.runGetState ((,) <$> S.getWord8 <*> S.getWord16be) bs 0
+        ((header, bitmap), rest) = case res of
+            Right v -> v
+            -- This happens only when bs is 23 bits or less.
```

Comment:
```suggestion
            -- This happens only when bs is 23 bytes or less.
```

Reply:
No, 3 bytes right?

Reply:
Yes, 3 bytes.  Better to use bytes than bits since that is what we are reading.